### DISCUSSION TITLE
Pruning dangling entries

### DISCUSSION THREAD

**Discussion by lexisother** [craftcms/cms#15765] [Q&A] [answered]
If I have a site with entries in various sections, and I delete the site by removing it from the project config and running `project-config/apply`, the entries will stay in the database and `gc/run` will not delete them. I know this is a rather unconventional way to delete a site (typically you also delete all related content through the UI) but I am debugging a multi-site setup and `git checkout -- config` is the easiest way to reset the site to the pre-modified state.

Is there any way to clean these dangling entries up from the database?

**Comment by elivz** [comment]
I have the same question. Looks like in Craft 5 you can just delete content using `DELETE FROM elements_sites WHERE siteId = x`. But I am not sure if that is going to leave other cruft in other tables.

**Comment by brandonkelly** [comment, marked as answer]
Sites only get soft-deleted initially, and won’t be fully deleted until `gc` is run after the [softDeleteDuration](https://craftcms.com/docs/5.x/reference/config/general.html#softdeleteduration) has passed (30 days by default).

To hard-delete the data immediately, you can run `craft gc --delete-all-trashed`; or answer `yes` to the “Delete all trashed items?” prompt when running `gc` interactively.
